### ISSUE TITLE
Querying aliases feature

### ISSUE THREAD

**Issue by rocketspacer** [Automattic/mongoose#5321]
Following #5184, I now want to implement the ability to query (and update) based on aliased fields. I've been looking at the code for a while. At first, I was looking for ways apply conditions aliasing on each and every model query. But then, I came up with another solution, which looks something like:  
```
var Foo = mongoose.Model('Foo', FooSchema);

Foo.findOneAndUpdate(
    Foo.translateAliases({aliasedField: ...}),
    Foo.translateAliases({aliasedField: ...})
)
```
Which I think is neater, clearer, and obviously optional. Plus, this won't change our codebase too much, avoiding cases when we forgot to do conditions aliasing in some queries.

I want to know if you feel okay going down this approach, so I put it here for feedback

**Comment by vkarpov15** [comment]
`translateAliases()` would definitely be a great step towards making aliases work seamlessly. If that function works, we can easily add it as a hook and then have it work transparently. Good idea :+1:

**Comment by wong2** [comment]
any progress?

**Comment by vkarpov15** [comment]
Nope not yet

**Comment by rafaelcorreiapoli** [comment]
Anu updates on this?

**Comment by vkarpov15** [comment]
@rafaelcorreiapoli no, not yet. Will bump this up for the next minor version.

**Comment by vkarpov15** [comment]
I didn't quite read this thread carefully, it looks like this was fixed by #5338. [Here's the docs for `Model.translateAlias()`](https://mongoosejs.com/docs/api/model.html#model_Model.translateAliases) @rafaelcorreiapoli
